### PR TITLE
Migrate messageCreate off sapphire

### PR DESCRIPTION
**Pull request by KevinBacabac**
## Summary of Changes
Migrates messageCreate logic to a discord.js based file in src/events.

## PRs
Resolves https://github.com/uwcsc/codeybot/issues/315

## Steps to Reproduce
Behaviour is confirmed to be the same with no regressions.

### CODE REVIEW

File: src/bot.ts [hunk 1]
```diff
@@ -44,6 +46,9 @@ export const startBot = async (): Promise<void> => {
     });
     client.on('error', client.logger.error);
     client.on('ready', initReady);
+    client.on('messageCreate', (message: Message) => {
+      if (client.user) initMessageCreate(client, container.logger, message);
+    });
     client.login();
```
Comment (KevinBacabac) [inline src/bot.ts:52]: Move if(client.user) check out

File: src/events/messageCreate.ts [hunk 2]
```diff
@@ -1,6 +1,5 @@
-import { ApplyOptions } from '@sapphire/decorators';
-import { container, Listener } from '@sapphire/framework';
-import { Message } from 'discord.js';
+import { ILogger } from '@sapphire/framework';
```
Comment (KevinBacabac) [inline src/events/messageCreate.ts:1]: Keep sapphire dependency for logger

File: src/events/messageCreate.ts [hunk 3]
```diff
@@ -61,7 +59,7 @@ const punishSpammersAndTrolls = async (message: Message): Promise<boolean> => {
           error: (err as Error).toString()
         });
       }
-      await sendKickEmbed(message, user, reason, isSuccessful);
+      await sendKickEmbed(client, message, user, reason, isSuccessful);
```
Comment (KevinBacabac) [inline src/events/messageCreate.ts:62]: Pass in the client as argument rather than using container

File: src/events/messageCreate.ts [hunk 4]
```diff
@@ -97,34 +95,23 @@ const convertResumePdfsIntoImages = async (message: Message): Promise<Message<bo
...
-
-    if (!client.user) {
-      return;
-    }
-
-    // Ignore all bots including self but not IRC
-    if (message.author.bot && message.author.id !== IRC_USER_ID) {
-      return;
-    }
+export const initMessageCreate = async (client: Client, logger: ILogger, message: Message): Promise<void> => {
```
Comment (KevinBacabac) [inline src/events/messageCreate.ts:98]: Convert class to function

File: src/utils/embeds.ts [hunk 5]
```diff
@@ -9,8 +8,13 @@ export const EMBED_COLOUR = '#0099ff';
 /*
  * Send kick embed
  */
-export const sendKickEmbed = async (message: Message, user: User, reason = '', isSuccessful = true): Promise<void> => {
-  const { client } = container;
```
Comment (KevinBacabac) [inline src/utils/embeds.ts:13]: Use client argument instead of container

File: src/bot.ts (hunk 1 above)
Comment (botatooo) [reply src/bot.ts:52]: Can't you just delete it

Comment (KevinBacabac) [reply src/bot.ts:52]: Good question for the devs I don't see why we have it but I copied it anyways

File: src/bot.ts [hunk 6]
```diff
@@ -44,6 +46,9 @@ export const startBot = async (): Promise<void> => {
     });
     client.on('error', client.logger.error);
     client.on('ready', initReady);
+    client.on('messageCreate', (message: Message) => {
+      if (client.user) initMessageCreate(client, container.logger, message);
```
Comment (mcpenguin) [inline src/bot.ts:50]: Use curly brackets for the `if` - imo it looks cleaner

**Review by mcpenguin — CHANGES_REQUESTED** [review]
small nitpick

File: src/bot.ts (hunk 6 above)
Comment (Picowchew) [reply src/bot.ts:50]: In actuality, if there are no linter warnings for something along the lines of `client.user` potentially being undefined, then you should be good to remove `if (client.user)`.

**Review by Picowchew — CHANGES_REQUESTED** [review]
Left a comment about `if (client.user)`.

**Review by Picowchew — APPROVED** [review]
LGTM!

**Review by mcpenguin — APPROVED** [review]
LGTM!

**Pull request merged**